src/lib/pgen.rs

```diff
@@ -344,3 +360,127 @@ pub fn goto1(grm: &Grammar, firsts: &HashMap<String, HashSet<String>>,
     closure1(grm, firsts, &mut goto);
     goto
 }
+
+// Haven't decided yet if I want to use a struct or not
+// struct State {
+//     id: usize,
```

Comment:
I was thinking at one point to store the unique_ids (eg. state numbers) inside state. But now their id is just their position in the `states`.